Infrastructure/PluginExecutor.cs

```diff
@@ -42,16 +49,24 @@ public void Run(string message, string user)
                     || CheckIfIlegalPluginPressed(message, out int pluginNumber, out msgForUser))
                 {
                     _messageSender.SendMessage(user, msgForUser);
-                    return;
+                    return ;
                 }
                 var extraData = ExtractExtraData(message);
                 var pluginType = ExtractPluginType(pluginNumber);
-
                 Execute(pluginType, extraData, user);
+                currentPluginId = _dal.LoadPluginData(user, SESSION_PLUGIN_ID);
             }
             else
             {
-                Execute(currentPluginId, message, user);
+                if (message.ToLower() == "main" || message.ToLower() == "home")
```

Comment:
You auto exit a session on this 2 keywords
Why so?
Maybe there is a plugin so ask user to type this keywords to do something?
If you override the functionality, it may make the system rigid :(

Reply:
ok i removed it